src/ui/control/logo_control.js

```diff
@@ -32,6 +32,7 @@ class LogoControl {
         this._container.appendChild(anchor);
         this._container.style.display = 'none';
 
+        this._map.on('styledata', this._updateLogo);
         this._map.on('sourcedata', this._updateLogo);
```

Comment:
I think you are safe to remove this listener all together. Keeping it here and removing the `metadata` condition causes `_updateLogo` to be called on every tile load 🙃

Reply:
I think this might still be necessary to ensure that the control updates when TileJSON is loaded.

Reply:
good catch, thank you! it was working on the debug page because the spritesheet was loading after the tilejson.